var/spack/repos/builtin/packages/zstd/package.py

```diff
@@ -55,6 +59,34 @@ class Zstd(MakefilePackage):
     # (last tested: nvhpc@22.3)
     conflicts("+programs %nvhpc")
 
+    build_system("cmake", "makefile", default="cmake")
+
+
+class CMakeBuilder(CMakeBuilder):
+    @property
+    def root_cmakelists_dir(self):
+        return os.path.join(super().root_cmakelists_dir, "build", "cmake")
+
+    def cmake_args(self):
+        spec = self.spec
+        args = []
+        args.append(self.define_from_variant("ZSTD_BUILD_PROGRAMS", "programs"))
+        args.extend(
+            [
+                self.define("ZSTD_BUILD_STATIC", self.spec.satisfies("libs=static")),
+                self.define("ZSTD_BUILD_SHARED", self.spec.satisfies("libs=shared")),
+            ]
+        )
+        if "compression=zlib" in spec:
+            args.append(self.define("ZSTD_ZLIB_SUPPORT", True))
+        if "compression=lzma" in spec:
+            args.append(self.define("ZSTD_LZMA_SUPPORT", True))
+        if "compression=lz4" in spec:
+            args.append(self.define("ZSTD_LZ4_SUPPORT", True))
+        return args
+
+
+class MakefileBuilder(MakefileBuilder):
```

Comment:
I think `build(self, spec, prefix)` below should have been `build(self, pkg, spec, prefix)`. Same for `install`?

```
==> Installing zstd-1.5.2-jjvcrhvi2y767kcrafqogmq7rg7elvdq
==> No binary for zstd-1.5.2-jjvcrhvi2y767kcrafqogmq7rg7elvdq found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/f7/f7de13462f7a82c29ab865820149e778cbfe01087b3a55b5332707abf9db4a6e.tar.gz
==> No patches needed for zstd
==> zstd: Executing phase: 'edit'
==> zstd: Executing phase: 'build'
==> Error: TypeError: build() takes 3 positional arguments but 4 were given
See build log for details:
  /tmp/rmeli/spack-stage/spack-stage-zstd-1.5.2-jjvcrhvi2y767kcrafqogmq7rg7elvdq/spack-build-out.txt

==> Error: zstd-1.5.2-jjvcrhvi2y767kcrafqogmq7rg7elvdq: Package was not installed
==> Error: Installation request failed.  Refer to reported errors for failing package(s).
```

Reply:
Yeah, also I'd like to undo cmake as a default since it's too big of a dependency. See #36606 (which is the second time I have to fix this lol, #25567 @johnwparent add a conflict if you need it)

Reply:
If this is the second time (sorry about that, was unaware it was an issue), perhaps another PR linking that issue as a comment above the builder designation so developers like myself who are out of the loop do not continue to make this mistake?

And no need for a conflict, `makefiles` are conflicted with Windows as a platform at the package level, so this doesn't really matter from a Windows perspective, we're getting CMake anyway.

Reply:
I just missed the PR, no worries